karyon3-archaius2/src/main/java/com/netflix/karyon/archaius/ArchaiusKaryonSuite.java

```diff
@@ -0,0 +1,229 @@
+package com.netflix.karyon.archaius;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.multibindings.MapBinder;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import com.netflix.archaius.Config;
+import com.netflix.archaius.config.CompositeConfig;
+import com.netflix.archaius.config.DefaultSettableConfig;
+import com.netflix.archaius.config.EnvironmentConfig;
+import com.netflix.archaius.config.MapConfig;
+import com.netflix.archaius.config.SettableConfig;
+import com.netflix.archaius.config.SystemConfig;
+import com.netflix.archaius.exceptions.ConfigException;
+import com.netflix.archaius.guice.ArchaiusModule;
+import com.netflix.archaius.guice.ConfigSeeder;
+import com.netflix.archaius.guice.ConfigSeeders;
+import com.netflix.archaius.guice.RootLayer;
+import com.netflix.archaius.inject.ApplicationLayer;
+import com.netflix.archaius.inject.DefaultsLayer;
+import com.netflix.archaius.inject.LibrariesLayer;
+import com.netflix.archaius.inject.RemoteLayer;
+import com.netflix.archaius.inject.RuntimeLayer;
+import com.netflix.governator.auto.AbstractPropertySource;
+import com.netflix.governator.auto.PropertySource;
+import com.netflix.karyon.AbstractKaryonSuite;
+import com.netflix.karyon.KaryonSuite;
+import com.netflix.karyon.ServerContext;
+
+/**
+ * Suite used to configure Archaius2 using a builder and making it available
+ * as a Governator PropertySource for conditional module loading as well as
+ * enabling it in the application.
+ * 
+ * @author elandau
+ *
+ */
+public abstract class ArchaiusKaryonSuite extends AbstractKaryonSuite {
+    private static final String DEFAULT_CONFIG_NAME     = "application";
+    
+    private static final String RUNTIME_LAYER_NAME      = "RUNTIME";
+    private static final String REMOTE_LAYER_NAME       = "REMOTE";
+    private static final String SYSTEM_LAYER_NAME       = "SYSTEM";
+    private static final String ENVIRONMENT_LAYER_NAME  = "ENVIRONMENT";
+    private static final String APPLICATION_LAYER_NAME  = "APPLICATION";
+    private static final String LIBRARIES_LAYER_NAME    = "LIBRARIES";
+    private static final String DEFAULTS_LAYER_NAME     = "DEFAULTS";
+    
+    static {
+        System.setProperty("archaius.default.configuration.class",      "com.netflix.archaius.bridge.StaticAbstractConfiguration");
+        System.setProperty("archaius.default.deploymentContext.class",  "com.netflix.archaius.bridge.StaticDeploymentContext");
+    }
+
+    public static class Builder {
```

Comment:
I think the hierarchy buildup can be done with empty properties at the very beginning as part of Karyon/KaryonBuilder construction, then we can get away from constructing a Karyon from inside configuration class? The bigger reason I suggest doing is that now the dynamic configuration is added in a different "addProfiles" step, I thought that will cause confusion for people.